### PR TITLE
Wrap tasks cmds as Actions for telemetry

### PR DESCRIPTION
**Pull request by Lixire**
- previously, we were unable to tell if cmds like `workbench.action.tasks.runTask` were accessed from the menu
- `Action` has automatic telemetry that adds a 'from' dimension that tells us where it was run
- thus, I wrapped the task commands as `Action` and called those at the task top-level menu

/cc @dbaeumer

### CODE REVIEW

File: src/vs/workbench/parts/tasks/common/taskActions.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+export class RunTaskAction extends Action {
+
+	public static ID = 'workbench.action.taskAction.runTask';
+	public static LABEL = nls.localize('runTaskAction', "Run task");
+
+	constructor(
+		id: string,
+		label: string,
+		@IQuickOpenService private quickOpenService: IQuickOpenService,
```
Comment (roblourens) [inline src/vs/workbench/parts/tasks/common/taskActions.ts]: Clean up unneeded services

File: src/vs/workbench/electron-browser/workbench.main.ts [hunk 2]
```diff
@@ -78,6 +78,7 @@ import 'vs/workbench/parts/trust/electron-browser/unsupportedWorkspaceSettings.c
 import 'vs/workbench/parts/relauncher/electron-browser/relauncher.contribution';
 
 import 'vs/workbench/parts/tasks/electron-browser/task.contribution';
+import 'vs/workbench/parts/tasks/common/taskActions';
```
Comment (roblourens) [inline src/vs/workbench/electron-browser/workbench.main.ts]: I think this should be in task.contribution.ts, which pulls in other task-related files.

**Comment by dbaeumer** [comment]
@roblourens @Lixire  can you PLEASE undo that and find a different solution for it. I spent quite some time to get rid of actions since they don't allow key binding tricks. That is why I converted everything to command. The telemetry that is currently logged for tasks has a trigger field telling you whether the command was triggered using Keyboard short cuts or the quick pick. We should extend this to have another tigger for global menu.

**Comment by dbaeumer** [comment]
I discussed this with @jrieken and instead of adding actions we could add telemetry that in general logs when a menu item is executed here: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/electron-browser/window.ts#L209

**Comment by jrieken** [comment]
Yeah, idea is to stop sprinkling telemetry information like `{source: menu}` or `{source: keyboard}` across the *whole* system but utilise the fact that there is usually a single that invokes commands from a source, e.g there is just [one](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/electron-browser/window.ts#L209) place from which global menus invoke commands (note that actions count as commands), also there is just [one](https://github.com/Microsoft/vscode/blob/master/src/vs/platform/keybinding/common/abstractKeybindingService.ts#L161) place from which a keybinding invokes commands, etc, etc

**Comment by roblourens** [comment]
Sure - most menu items will send two telemetry events then, their workbenchActionExecuted and also this new event. But it makes sense to add it in a single place.

**Comment by dbaeumer** [comment]
Just some clarification since I left it out in my comment here: https://github.com/Microsoft/vscode/pull/29223#issuecomment-310298815

Having tasks as actions and commands doubles them in the UI for commands (F1) and the Keybindings editor.

capture


This can TMK only be avoided by using either only actions or commands. Switching back to actions is quite some work and it will break a feature that relies on commands: assigning a keyboard shortcut to a specific task.

**Comment by jrieken** [comment]
With https://github.com/Microsoft/vscode/commit/a9aa12d1221572abec33517a226a88f72b1bf0d4 onwards the changes here shouldn't be needed anymore

**Comment by Lixire** [comment]
Gotcha, I'll remove it today.

**Pull request merged**